src/commands/check.js

```diff
@@ -76,7 +76,8 @@ export const handler = catchAsyncError(async opts => {
     .map(({ncuValue}) => ncuValue)
     .join(',');
   const currentVersions = ncu.getCurrentDependencies(packageJson, {dep: ncuDepGroups});
-  const latestVersions = await ncu.queryVersions(currentVersions, {versionTarget: 'latest'});
+  const peerDependencies = ncu.getPeerDependencies(currentVersions, {});
+  const latestVersions = await ncu.queryVersions(currentVersions, {versionTarget: 'latest', peerDependencies});
```

Comment:
Is there a way to show some explanation to the user e.g. that latest released version of the dependency is v2 but the latest version that satisfies `peerDependencies` is v1.2?

Reply:
Currently, with `npm-check-updates`, you can provide only one strategy at time. Hence, we would have to call `ncu.queryVersions` two times (one call with peerDependencies and one call without them). However, the package data will then be fetched two times from npm servers which doubles the process time. I don't think that this would be a good solution.

Reply:
> which doubles the process time

We can fetch them in parallel. My concern here is it may introduce confusion when you're absolutely sure the latest version of the package is `v2` but `npm-upgrade` tells you that the latest version is `v1.2` without any explanation.

By the way, what does `ncu.getPeerDependencies(currentVersions, {})` method actually do?

Reply:
> We can fetch them in parallel. My concern here is it may introduce confusion when you're absolutely sure the latest version of the package is `v2` but `npm-upgrade` tells you that the latest version is `v1.2` without any explanation.

I understand that and I just had a deeper insight in the code. It looks like `npm-check-updates` caches the network responses, so a second call of `ncu.queryVersions()` doesn't trigger any new network requests and hence runs quite fast. Therefore, we can call it indeed twice, without any problems.

I just pushed a new commit which adds a hint, if the suggested version is not the latest version. It prints `(ignored incompatible latest version X.Y.Z)` in the output table. How about that?

-----

> By the way, what does `ncu.getPeerDependencies(currentVersions, {})` method actually do?

For every package in `currentVersions`, it reads the local `package.json` (from the directory `node_modules/package/`) and extracts the section `peerDependencies`. All results are merged and sorted by package. See https://github.com/raineorshine/npm-check-updates/blob/v11.4.1/lib/index.js#L232

The result is added to the options, so that the [`filterPredicate`](https://github.com/raineorshine/npm-check-updates/blob/v11.4.1/lib/package-managers/npm.js#L186) can filter all versions that satisfy the peer dependencies.

Reply:
Okay, I've changed the output for incompatible updates. Here are two examples:

## First example
```
Checking for outdated dependencies for ".../package.json"...
[====================] 2/2 100%

New versions of active modules available:

  ncu-test-return-version   1.0.0   →   1.1.0 

Ignored incompatible updates (peer dependencies):

  ncu-test-return-version   1.0.0   →   2.0.0 

```

## Second example
```
Checking for outdated dependencies for ".../package.json"...
[====================] 50/50 100%

New versions of active modules available:

  @nextcloud/dialogs                ^3.1.1   →     ^3.1.2 
  @nextcloud/router                 ^1.2.0   →     ^2.0.0 
  @nextcloud/vue                    ^3.8.0   →     ^3.9.0 
  @nextcloud/webpack-vue-config     ^4.0.1   →     ^4.0.2 
  @babel/core                     ^7.13.14   →   ^7.13.15 
  @babel/preset-env               ^7.13.12   →   ^7.13.15 
  core-js                          ^3.10.0   →    ^3.10.1 
  eslint                           ^7.23.0   →    ^7.24.0 
  eslint-plugin-vue                 ^7.8.0   →     ^7.9.0 
  webpack                          ^5.28.0   →    ^5.32.0 

Ignored incompatible updates (peer dependencies):

  css-loader               ^4.3.0   →    ^5.2.1 
  eslint-plugin-promise    ^4.3.1   →    ^5.1.0 
  sass-loader             ^10.1.1   →   ^11.0.1 

Ignored updates:

                  From          To   Ignored versions   Reason        
  vue-fragment   1.5.1   →   1.5.2        1.5.2         buggy version 

```